lib/iris/tests/test_io_init.py

```diff
@@ -12,40 +12,42 @@
 import iris.tests as tests  # isort:skip
 
 from io import BytesIO
-import pathlib
+from pathlib import Path
 
 import iris.fileformats as iff
 import iris.io
 
 
 class TestDecodeUri(tests.IrisTest):
-    def test_decode_uri(self):
+    def test_decode_uri__str(self):
         tests = {
-            "/data/local/someDir/PP/COLPEX/COLPEX_16a_pj001.pp": (
+            (uri := "/data/local/someDir/PP/COLPEX/COLPEX_16a_pj001.pp"): (
                 "file",
-                "/data/local/someDir/PP/COLPEX/COLPEX_16a_pj001.pp",
+                uri,
             ),
-            r"C:\data\local\someDir\PP\COLPEX\COLPEX_16a_pj001.pp": (
+            (uri := r"C:\data\local\someDir\PP\COLPEX\COLPEX_16a_pj001.pp"): (
                 "file",
-                r"C:\data\local\someDir\PP\COLPEX\COLPEX_16a_pj001.pp",
+                uri,
             ),
-            "file:///data/local/someDir/PP/COLPEX/COLPEX_16a_pj001.pp": (
-                "file",
-                "///data/local/someDir/PP/COLPEX/COLPEX_16a_pj001.pp",
+            (
+                uri := "file:///data/local/someDir/PP/COLPEX/COLPEX_16a_pj001.pp"
+            ): (
+                uri[:4],
+                uri[5:],
             ),
-            "http://www.somehost.com:8080/resource/thing.grib": (
-                "http",
-                "//www.somehost.com:8080/resource/thing.grib",
+            (uri := "http://www.somehost.com:8080/resource/thing.grib"): (
+                uri[:4],
+                uri[5:],
             ),
-            "/data/local/someDir/2013-11-25T13:49:17.632797": (
+            (uri := "/data/local/someDir/2013-11-25T13:49:17.632797"): (
                 "file",
-                "/data/local/someDir/2013-11-25T13:49:17.632797",
+                uri,
             ),
         }
-        for uri, pair in tests.items():
-            self.assertEqual(pair, iris.io.decode_uri(uri))
+        for uri, expected in tests.items():
+            self.assertEqual(expected, iris.io.decode_uri(uri))
 
-    def test_decode_uri_path_object(self):
+    def test_decode_uri__path_object(self):
```

Comment:
```suggestion
    def test_decode_uri__path(self):
```
Fine with you ignoring this, would just make it slightly more consistent imo